### PR TITLE
[DOCS] How to pass an in-memory DataFrame to a Checkpoint

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- GREAT-356/GREAT-378
- Documentation for passing in a DataFrame to a Checkpoint using a RuntimeBatchRequest

Out of scope:
- Checkpoint configuration precedence behavior when yaml configuration conflicts with runtime configuration

### Definition of Done
- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

**Review by alexsherstinsky — APPROVED** [review]
LGTM

File: great_expectations/data_context/data_context.py
Comment (Shinnnyshinshin) [inline great_expectations/data_context/data_context.py:3261]: ❤️

**Review by Shinnnyshinshin — APPROVED** [review]

**Review by Shinnnyshinshin — CHANGES_REQUESTED** [review]
Looks wonderful to me too. Before you merge, could I ask that a link to the script be added to the bottom of the Doc? It's usually placed under **Additional Notes**, and you should be able to find plenty of examples in the `Connect to your Data` section

https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/filesystem/pandas

**Comment by NathanFarmer** [comment]
@Shinnnyshinshin 
> Looks wonderful to me too. Before you merge, could I ask that a link to the script be added to the bottom of the Doc?

✅ You will not have to ask me to do this a 3rd time 😄

**Review by Shinnnyshinshin — APPROVED** [review]
Looks great :)

**Review by anthonyburdi — APPROVED** [review]
LGTM! (pending resolution of the integration test failure)

**Review by donaldheppner — APPROVED** [review]
sidebar looks good

**Pull request merged**
